File: B/B_TrickTaking.go
package main

import (
	"bufio"
	"fmt"
	"io"
	"os"
	"strconv"
)

var sc *bufio.Scanner

func readInt() int {
	sc.Scan()
	i, _ := strconv.Atoi(sc.Text())
	return i
}

func run(stdin io.Reader, out io.Writer) {
	sc = bufio.NewScanner(stdin)
	sc.Split(bufio.ScanWords)

	N, T := readInt(), readInt()

	C := make([]int, N)
	for i := range C {
		C[i] = readInt()
	}
	R := make([]int, N)
	for i := range R {
		R[i] = readInt()
	}
	var ans int = checkT(C, R, T)
	if ans != -1 {
		fmt.Print(ans)
	} else {
		ans = checkOne(C[0], R[0], C, R)
		fmt.Print(ans)
	}
}

func main() {
	run(os.Stdin, os.Stdout)
}

func checkT(C []int, R []int, T int) int {
	var ans int = -1
	var value int = -1
	for i := 0; i < len(C); i++ {
		if C[i] == T && R[i] > value {
			value = R[i]
			ans = i + 1
		}
	}
	return ans
}

func checkOne(C1 int, R1 int, C []int, R []int) int {
	var ans int = -1
	var value int = R1
	for i := 0; i < len(C); i++ {
		if C[i] == C1 {
			if R[i] > value {
				value = R[i]
				ans = i + 1
			}
		}
	}
	if ans != -1 {
		return ans
	} else {
		return 1
	}
}

File: src/B_TrickTaking.java
import java.util.*;

public class B_TrickTaking  {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
N
        int N = scanner.nextInt();
        int T = scanner.nextInt();
        int C[] = new int[N];
        int R[] = new int[N];
        for (int i = 0; i < N; i++) {
            C[i] = scanner.nextInt();
        }
        for (int i = 0; i < N; i++) {
            R[i] = scanner.nextInt();
        }
        scanner.close();

        int ans = checkT(C, R, T);
        if (ans != -1) {
            System.out.println(ans);
        } else {
            ans = checkOne(C[0], R[0], C, R);
            System.out.println(ans);
        }
    }

    private static int checkT(int[] C, int[] R, int T) {
        int ans = -1;
        int value = -1;
        for (int i = 0; i < C.length; i++) {
            if (C[i] == T && R[i] > value) {
                value = R[i];
                ans = i+1;
            }
        }
        return ans;
    }

    private static int checkOne(int C1, int R1, int[] C, int[] R) {
        int ans = -1;
        int value = R1;
        for (int i = 1; i < C.length; i++) {
            if (C[i] == C1) {
                if (R[i] > value) {
                    value = R[i];
                    ans = i+1;
                }
            }
        }
        if (ans != -1) {
            return ans;
        } else {
            return 1;
        }
    }
}

